realm/realm-library/src/main/cpp/jni_util/java_local_ref.hpp

```diff
@@ -46,11 +46,6 @@ class JavaLocalRef {
         m_env->DeleteLocalRef(m_jobject);
     }
 
-    JavaLocalRef(const JavaLocalRef&) = delete;
-    JavaLocalRef& operator=(const JavaLocalRef&) = delete;
-    JavaLocalRef(JavaLocalRef&& rhs) = delete;
-    JavaLocalRef& operator=(JavaLocalRef&& rhs) = delete;
```

Comment:
The default move/copy constructor won't work. So remove those lines will cause crash when you copy/move a non-null `JavaLocalRef`. (Now it doesn't crash just because of you copy a non-null `JavaLocalRef` to a null one)

So if the copy/move constructor is needed, then it needs the implementation as well.